alloc is required for wifi?

Steps to reproduce:

```
$ esp-generate --chip esp32 -o wifi myproject
$ cd myproject
$ cargo check
…
   Compiling esp-hal-procmacros v0.15.0
   Compiling xtensa-lx-rt v0.17.2
error[E0463]: can't find crate for `alloc`
  --> /home/mart-e/.cargo/registry/src/index.crates.io-6f17d22bba15001f/esp-wifi-0.11.0/src/lib.rs:84:1
   |
84 | extern crate alloc;
   | ^^^^^^^^^^^^^^^^^^^ can't find crate

For more information about this error, try `rustc --explain E0463`.
The following warnings were emitted during compilation:

warning: esp-wifi@0.11.0: esp-wifi should be built with optimization level 2 or 3 - yours is s.

error: could not compile `esp-wifi` (lib) due to 1 previous error
```

Executing `esp-generate --chip esp32 -o alloc -o wifi myproject` works. I don’t know what `alloc` does but it seems it is needed when enabling `wifi`.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Update the resulting binary name (#62)
 
 ### Fixed
+- Verify the required options are provided (#65)
 
 ### Removed
 

--- a/src/main.rs
+++ b/src/main.rs
@@ -83,6 +83,13 @@ impl GeneratorOptionItem {
             GeneratorOptionItem::Option(option) => option.chips,
         }
     }
+
+    fn enables(&self) -> &[&str] {
+        match self {
+            GeneratorOptionItem::Category(_) => &[],
+            GeneratorOptionItem::Option(option) => option.enables,
+        }
+    }
 }
 
 static OPTIONS: &[GeneratorOptionItem] = &[
@@ -466,6 +473,18 @@ fn process_options(args: &Args) {
                     "Option '{}' is not supported for chip {}",
                     option,
                     args.chip
+                );
+                process::exit(-1);
+            }
+            if !option_item
+                .enables()
+                .iter()
+                .all(|requirement| args.option.contains(&requirement.to_string()))
+            {
+                log::error!(
+                    "Option '{}' requires {}",
+                    option_item.name(),
+                    option_item.enables().join(", ")
                 );
                 process::exit(-1);
             }

